Java/Concentus/src/main/java/org/concentus/OpusEncoder.java

```diff
@@ -1327,6 +1326,32 @@ public int encode(short[] in_pcm, int pcm_offset, int frame_size,
         }
     }
 
+    /**
+     * Encodes an Opus frame, putting the output into a specified data buffer
+     * @param in_pcm 16-bit input signal (Interleaved if stereo), in a byte array. Length should be at least frame_size * channels * 2
+     * @param pcm_offset Offset to use when reading the in_pcm buffer
+     * @param frame_size The number of samples _per channel_ in the inpus signal. The frame size must be a valid Opus framesize for the given sample rate.
+     * For example, at 48Khz the permitted values are 120, 240, 480, 960, 1920, and 2880. Passing in a duration of less than 10ms
+     * (480 samples at 48Khz) will prevent the encoder from using FEC, DTX, or hybrid modes.
+     * @param out_data Destination buffer for the output payload. This must contain at least max_data_bytes
+     * @param out_data_offset The offset to use when writing to the output data buffer
+     * @param max_data_bytes The maximum amount of space allocated for the output payload. This may be used to impose
+     * an upper limit on the instant bitrate, but should not be used as the only bitrate control (use setBitrate for that)
+     * @return The length of the encoded packet, in bytes
+     * @throws OpusException 
+     */
+    public int encode(byte[] in_pcm, int pcm_offset, int frame_size,
+            byte[] out_data, int out_data_offset, int max_data_bytes) throws OpusException {
+    	//Convert byte array to short array
+    	short[] spcm = new short[in_pcm.length / 2];
+		for (int c = 0; c < spcm.length; c++) {

```

Comment:
you need to take into account pcm_offset here and then pass pcm_offset=0 on line 1352